share/wake/lib/core/json.wake

```diff
@@ -13,161 +13,196 @@
+    rec Nil (if indent > 0 then Some "" else None) lhs
+
 
 export def defaultJSONFormat: JSONFormat =
-  def formatDouble d =
-    match (dclass d)
-      DoubleInfinite if d <. 0e0 = "-Infinity"
-      DoubleInfinite = "Infinity"
-      DoubleNaN      = "NaN"
-      _              = dstr d
-  JSONFormat jsonEscape str formatDouble 0
+    def formatDouble d = match (dclass d)
+        DoubleInfinite if d <. 0e0 = "-Infinity"
+        DoubleInfinite = "Infinity"
+        DoubleNaN = "NaN"
+        _ = dstr d
+    JSONFormat jsonEscape str formatDouble 0
+
 
 export def prettyJSONFormat = defaultJSONFormat | setJSONFormatIndent 2
 
+
 export def customFormatJSON fmt body = doFormat fmt body | cat
+
+
 export def formatJSON = customFormatJSON defaultJSONFormat
+
+
 export def prettyJSON = customFormatJSON prettyJSONFormat
 
+
 export def root /| filterFn = jfilter filterFn root
+
+
 export def jfilter filterFn root = match root
-  JArray l = JArray (filter filterFn l)
-  _ = if filterFn root then JArray (root, Nil) else JArray Nil
+    JArray l = JArray (filter filterFn l)
+    _ = if filterFn root then JArray (root, Nil) else JArray Nil
+
 
 export def root /../ filterFn = jfind filterFn root
+
+
 export def jfind filterFn root =
-  def helper node acc = match node
-    JArray l =
-      def tail = foldr helper acc l
-      if filterFn node then node, tail else tail
-    JObject l =
-      def tail = foldr (helper _.getPairSecond _) acc l
-      if filterFn node then node, tail else tail
-    _ =
-      if filterFn node then node, acc else acc
-  helper root Nil | JArray
+    def helper node acc = match node
+        JArray l =
+            def tail = foldr helper acc l
+            if filterFn node then node, tail else tail
+        JObject l =
+            def tail = foldr (helper _.getPairSecond _) acc l
+            if filterFn node then node, tail else tail
+        _ = if filterFn node then node, acc else acc
+    helper root Nil | JArray
+
 
 export def jempty root = empty (jlist root)
+
+
 export def jlist root = match root
-  JArray x = x
-  _ = root, Nil
+    JArray x = x
+    _ = root, Nil
+
 
 export def x // y =
-  def helper tail = match _
-    JString  _ = tail
-    JInteger _ = tail
-    JDouble  _ = tail
-    JBoolean _ = tail
-    JNull      = tail
-    JObject  l =
-      def flatten v tail = match v.getPairSecond
-        JArray l = l ++ tail
-        w = w, tail
-      foldr flatten tail (filter (matches y _.getPairFirst) l)
-    JArray   l =
-      def flatten v tail = helper tail v
-      foldr flatten tail l
-  JArray (helper Nil x)
+    def helper tail = match _
+        JString _ = tail
+        JInteger _ = tail
+        JDouble _ = tail
+        JBoolean _ = tail
+        JNull = tail
+        JObject l =
+            def flatten v tail = match v.getPairSecond
+                JArray l = l ++ tail
+                w = w, tail
+            foldr flatten tail (filter (matches y _.getPairFirst) l)
```

Comment:
Matching `match` case line breaks does mean sections like this will have a lot of very short lines, but the parallel structures are probably still a net positive.